backend/molgenis-emx2-datamodels/src/main/java/org/molgenis/emx2/datamodels/profiles/SchemaFromProfile.java

```diff
@@ -59,23 +65,132 @@ public SchemaMetadata create() throws MolgenisException {
+    Map<String, List<Row>> columnDeclarationsByTableName = new HashMap<>();
     List<Row> keepRows = new ArrayList<>();
     String[] modelsList = getResourceListing(directory);
-    for (String schemaLoc : modelsList) {
 
+    for (String schemaLoc : modelsList) {
       Iterable<Row> rowIterable =
           CsvTableReader.read(
               new InputStreamReader(
                   Objects.requireNonNull(
                       getClass().getResourceAsStream(directory + File.separator + schemaLoc))));
-
       for (Row row : rowIterable) {
-        List<String> profiles = csvStringToList(row.getString("profiles"));
-        if (profiles.isEmpty()) {
-          throw new MolgenisException("No profiles for " + row);
+
+        String tableName = row.getString("tableName");
+        if (rowIsTableDeclaration(row)) {
+          if (tableDeclarationsByTableName.containsKey(tableName)) {
+            throw new MolgenisException("Duplicate table declaration for name: " + tableName);
+          }
+          tableDeclarationsByTableName.put(tableName, row);
+        } else {
+          if (!columnDeclarationsByTableName.containsKey(tableName)) {
+            columnDeclarationsByTableName.put(tableName, new ArrayList<>());
+          }
+          columnDeclarationsByTableName.get(tableName).add(row);
+        }
+      }
+    }
+
+    // post process: if table is tagged, but no columns, add tag to all columns
+    for (String tableName : tableDeclarationsByTableName.keySet()) {
+      if (columnDeclarationsByTableName.get(tableName) == null) {
+        // it is allowed to have tables without any columns, skip
+        continue;
+      }
+
+      Row row = tableDeclarationsByTableName.get(tableName);
+
+      List<String> profilesForTable = csvStringToList(row.getString("profiles"));
+
+      for (String profileForTable : profilesForTable) {
+        // table is tagged with selected profile, now check columns
+        boolean atLeastOneColumnIsTagged = false;
+
+        for (Row column : columnDeclarationsByTableName.get(tableName)) {
+
+          List<String> profilesForColumn = csvStringToList(column.getString("profiles"));
+          if (profilesForColumn.contains(profileForTable)) {
+            atLeastOneColumnIsTagged = true;
+            break;
+          }
         }
-        for (String profile : profiles) {
-          if (this.profiles.getProfileTagsList().contains(profile)) {
+        // tag all when none are tagged by this particular profileTag
+        if (!atLeastOneColumnIsTagged) {
+          for (Row column : columnDeclarationsByTableName.get(tableName)) {
+            List<String> profilesForColumn =
+                column.getString("profiles") != null
+                    ? csvStringToList(column.getString("profiles"))
+                    : new ArrayList<>();
+            profilesForColumn.add(profileForTable);
+            column.setString("profiles", String.join(",", profilesForColumn));
```

Comment:
Hmm, this means that import/export of the model would result in a change (having tags added that user didn't add), am I correct? If yes, I rather prevent that.